packages/server-wallet/src/wallet/index.ts

```diff
@@ -490,32 +491,44 @@ export class Wallet extends EventEmitter<WalletEvent>
 
     // TODO: Move into utility somewhere?
     function handleRequest(outbox: Outgoing[]): (req: ChannelRequest) => Promise<void> {
-      return async ({channelId}: ChannelRequest): Promise<void> => {
-        const {states: signedStates, channelState} = await store.getStates(channelId);
+      return async (req: ChannelRequest): Promise<void> => {
+        const {channelId} = req;
 
-        const {participants, myIndex} = channelState;
+        if (req.type === 'GetChannel') {
+          const {states: signedStates, channelState} = await store.getStates(channelId);
 
-        createOutboxFor(channelId, myIndex, participants, {signedStates}).map(outgoing =>
-          outbox.push(outgoing)
-        );
+          const {participants, myIndex} = channelState;
+
+          createOutboxFor(channelId, myIndex, participants, {signedStates}).map(outgoing =>
+            outbox.push(outgoing)
+          );
+        } else if (req.type === 'ProposeLedger') {
+          await store.storeTheirLedgerCommit(channelId, req.outcome);
+        }
       };
     }
 
     const {channelIds, objectives, channelResults: fromStoring} = await this.store.pushMessage(
       wirePayload
     );
 
-    const {channelResults, outbox} = await this.takeActions(channelIds);
+    let outbox: Outgoing[] = [];
+
+    if (wirePayload.requests && wirePayload.requests.length > 0)
+      // Modifies outbox, may append new messages
+      await Promise.all(wirePayload.requests.map(deserializeRequest).map(handleRequest(outbox)));
+
+    const runLoopResult = await this.takeActions(channelIds);
```

Comment:
Had to reverse the ordering of (1) take actions (2) handle requests. This might be the reason that using an objective is preferred, since objectives already have the ordering appropriate.